python_submitty_utils/submitty_utils/submitty_schema_validator.py

```diff
@@ -0,0 +1,186 @@
+        # validate each container in the testcase
+        for c in containers:
+            try:
+                validate_schema(c, c_schema, prefix=t_name, warn=warn)
+            except SubmittySchemaException:
+                raise
+
+        validators = t['validation']
+        validator_num = 0
+        # Validate each validation object in the testcase
+        for v in validators:
+            try:
+                validate_schema(v, abs_v_schema, prefix=t_name, warn=warn)
+                validate_schema(v, spec_v_schema, prefix=t_name, warn=warn)
+            except SubmittySchemaException:
+                raise
+            validator_num += 1
+        testcase_num += 1
+
+        try:
+            validate_schema(t, t_s, prefix=t_name, warn=warn)
+        except SubmittySchemaException:
+            raise
+    # Finally, validate the config as a whole.
+    try:
+        validate_schema(j_, s_, prefix='Your config json', warn=warn)
+    except SubmittySchemaException:
+        raise
+
+
+def validate_schema(j_, s_, key='', prefix='', required=True, warn=False):
+    """
+    Validate a configuration against a schema.
+
+    A function which validates a key within a config and a schema.
+    If no key is given, we evaluate the whole config or schema.
+    """
+    descriptive_title = '{0} {1}'.format(prefix, key)
+    if key != '':
+        my_json_chunk = j_.get(key, None) if j_ is not None else j_
+        my_schema = s_.get(key, None) if s_ is not None else s_
+    else:
+        my_json_chunk = j_
+        my_schema = s_
+    # If attempting to validate an item not specified in the schema, fail.
+    if my_schema is None:
+        raise SubmittySchemaException(j_,
+                                      s_,
+                                      ("ERROR: There is no specification for "
+                                       "{0} in the schema. Please "
+                                       "add a specification.").format(key),
+                                      descriptive_title,
+                                      None)
+    if my_json_chunk is None:
+        if warn:
+            print(("WARNING: could not identify "
+                   "{0} ({1})").format(descriptive_title, key))
+        return
+    try:
+        # validate.
+        validate(instance=my_json_chunk, schema=my_schema)
+    except Exception as e:
+        raise SubmittySchemaException(my_json_chunk,
+                                      my_schema,
+                                      ('ERROR: {0} was not properly '
+                                       'formatted')
+                                      .format(descriptive_title),
+                                      descriptive_title,
+                                      e)
+
+
+def validate_complete_config_schema_using_filenames(config_path,
+                                                    schema_path,
+                                                    show_warnings=True):
```

Comment:
Where is this going to be used where `show_warnings` is going to be True? It looks like all current usage it is set to `False`?

Reply:
Changed the default to be ```False```.